### ISSUE TITLE
endlines

### ISSUE THREAD

**Issue by evantarrh** [mayankmahajan24/QL#16]
i hate this

<img width="621" alt="screenshot 2015-12-02 03 16 42" src="https://cloud.githubusercontent.com/assets/5566223/11525683/2a909514-98a3-11e5-8f5a-df6908011d2b.png">

**Comment by mayankmahajan24** [comment]
what am i looking at

**Comment by evantarrh** [comment]
lol sorry–––ok so here is a program that we accept:

```
json testJson = json("test.json") 
[NEWLINE]
```

here are examples of programs that we reject:

```
json testJson = json("test.json")
[NEWLINE]
[NEWLINE]
```

```
json testJson = json("test.json")
```

**Comment by anshulkgupta** [comment]
Lol this is because of our decision to use ENDLINE instead of SEMICOLON to mark off lines. We could let ENDLINE be its own statement that does nothing and that should help us be a little more tolerant?

**Comment by mayankmahajan24** [comment]
How about we just take [\n]+ and read it in the scanner as ENDLINE.

And when we parse the program, we append an "\n" to the end of the program
always. If there are no ENDLINEs are the end, this will add one. If there
is 1 or more "\n"s already it would be read as just one ENDLINE.
Can something like this be achieved?

Mayank Mahajan

Columbia University | Class of 2017
B.S. Computer Science
mm4399@columbia.edu
www.linkedin.com/in/mmahajan24/
571.212.5535

**Comment by anshulkgupta** [comment]
I was about to open an issue on this just now and realized that we had one already. Let's get this in -- can someone get this in our next parser update?

**Comment by evantarrh** [comment]
@anshulkgupta yes i will do this when i fix the conflicts

**Comment by evantarrh** [comment]
Closed in #22.
